### DISCUSSION TITLE
⭐ Please, could you adjust groups?

### DISCUSSION THREAD

**Discussion by WanderFox** [Comfy-Org/ComfyUI#4874] [Ideas]
Currently, we can only resize groups from the bottom-right corner. Additionally, when multiple groups are selected, we can only move the nodes within the groups and not the groups themselves. It would be helpful to have the ability to resize groups from any corner and to move multiple groups along with their nodes

**Comment by jtk1996** [comment]
It would also be helpful (necessary?) to make groups real containers/objects, such that you cannot move nodes by accident to a different group when touching them. Which would also mean you might have to setup group in group features.

**Reply by webfiltered** [reply]
Agreed - mentioned this the other day.  As an interim, some basic logic was added to groups to make this _less_ of an issue (and kind of fake it) until it's implemented properly.

**Comment by LukeG89** [comment]
They are working on improving groups.

From 1.3.24 frontend you can already select multiple groups and drag them:

- https://github.com/Comfy-Org/litegraph.js/pull/225

And today they released Group nesting and Group membership in 1.3.32 nightly frontend:

- https://github.com/Comfy-Org/litegraph.js/pull/262

And you can also select and delete groups!

Here's what you can do in 1.3.32 frontend:

https://github.com/user-attachments/assets/0f5fe9a1-ebeb-4f6d-abe3-df00ba3d53da

**Reply by LukeG89** [reply]
> It would be helpful to have the ability to resize groups from any corner

@WanderFox I don't think they planned to do it.

You can open a feature request here: https://github.com/Comfy-Org/litegraph.js/issues

**Reply by webfiltered** [reply]
An issue would be appreciated.  There's framework currently going in that will make adding these features easier.

**Comment by jtk1996** [comment]
Sorry for late reply!
 
Thank you for the information and example, looks very promising!   


Gesendet: Montag, 4. November 2024 um 15:19
Von: LukeG89 ***@***.***>
An: comfyanonymous/ComfyUI ***@***.***>
CC: jtk1996 ***@***.***>,Comment ***@***.***>
Betreff: Re: [comfyanonymous/ComfyUI] ⭐ Please, could you adjust groups? (Discussion #4874)

 
Here's what you can do in 1.3.32 frontend:
https://github.com/user-attachments/assets/5f2bc876-cef0-4357-a3f5-388ad3bf9231

**Comment by jridev** [comment]
Just touching on this subject, it would be nice to have groups be collapsible like every other node.
